connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+            )
+            return False
+
+        if attachment_size > FILE_SIZE_LIMIT:
+            self._logger.warning(
+                f"File size {attachment_size} of file {attachment_name} is larger than {FILE_SIZE_LIMIT} bytes. Discarding file content"
+            )
+            return
+        return True
+
+    async def _get_document_with_content(self, doc):
+        document = {
+            "_id": doc["id"],
+            "_timestamp": doc["date_time"],
+        }
+
+        temp_filename = ""
+
+        async with NamedTemporaryFile(mode="wb", delete=False) as async_buffer:
+            response = await self.client.get_file_content(
+                download_url=doc["download_url"]
+            )
+            await async_buffer.write(response.encode("utf-8"))
+            temp_filename = str(async_buffer.name)
+
+        await asyncio.to_thread(
+            convert_to_b64,
+            source=temp_filename,
+        )
+        async with aiofiles.open(file=temp_filename, mode="r") as target_file:
+            # base64 on macOS will add a EOL, so we strip() here
+            document["_attachment"] = (await target_file.read()).strip()
+        try:
+            await remove(temp_filename)
+        except Exception as exception:
+            self._logger.warning(
+                f"Could not remove file: {temp_filename}. Error: {exception}"
+            )
+
+        return document
+
+    async def get_content(self, doc, timestamp=None, doit=False):
+        attachment_size = doc["file_size"]
+        if not (doit and attachment_size > 0):
+            return
+
+        attachment_name = doc["file_name"]
+
+        attachment_extension = (
+            attachment_name[attachment_name.rfind(".") :]  # noqa
+            if "." in attachment_name
+            else ""
+        )
+
+        if not self._pre_checks_for_get_content(
+            attachment_extension=attachment_extension,
+            attachment_name=attachment_name,
+            attachment_size=attachment_size,
+        ):
+            return
+
+        self._logger.debug(f"Downloading {attachment_name}")
+        return await self._get_document_with_content(doc)
+
+    async def fetch_previous_meeting_details(self, meeting_id):
+        previous_meeting = await self.client.get_past_meeting(meeting_id=meeting_id)
+
+        if not previous_meeting:
+            return
+
+        participants = []
+        async for participant in self.client.get_past_meeting_participants(
+            meeting_id=meeting_id
+        ):
+            participants.append(participant)
+
+        previous_meeting["participants"] = participants
+        return previous_meeting
+
+    async def fetch_recordings(self, user_id):
+        async for recording in self.client.get_recordings(user_id=user_id):
+            yield recording
+
+        async for recording in self.client.get_trash_recordings(user_id=user_id):
+            yield recording
```

Comment:
Is syncing trashed recordings a requirement? This strikes me as something that customers would probably not want by default. At the very least, I'd think this should be configurable. CC @danajuratoni

Reply:
In the future, it makes sense to make this configurable for all connectors with Trash/Recycle Bin function. 
Right now, we should stay consistent with the other connectors e.g. GDrive, Salesforce and Gmail which are ignoring trashed items and ignore trashed items for Zoom too.